### PR TITLE
add cuda-python extension demo

### PR DESCRIPTION
**Pull request by t-vi**

### CODE REVIEW

**Comment by t-vi** [comment]
@riccardofelluga this is the cuda-python example I had in mind.

**Comment by t-vi** [comment]
Crazy timeout... :frowning_face:

**Review by carmocca — APPROVED** [review]
LGTM! I played with the kernel in a studio for a bit. The error messages when you mess up are not great but that is probably not your fault.

```python
RuntimeError: NVRTC error: (<nvrtcResult.NVRTC_SUCCESS: 0>, b'NVRTC_ERROR_COMPILATION')
```

My comments are tiny irrelevant typos. I think the GitHub UI is not rendering them properly. Let me know and then maybe I can apply them manually

**Comment by carmocca** [comment]
cc @Andrei-Aksionov, you might have some fun with this too

File: notebooks/extend_thunder_with_cuda_python.ipynb [hunk 1]
```diff
@@ -0,0 +1,1095 @@
...
+    "hidePrompt": false
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "tensor(1.3113e-06)"
+      ]
+     },
+     "execution_cou
```
[the remark this answers is not in the document]
Comment (t-vi) [reply notebooks/extend_thunder_with_cuda_python.ipynb:767]: But this should work on proxies?

[the remark this answers is not in the document]
Comment (t-vi) [reply notebooks/extend_thunder_with_cuda_python.ipynb:767]: It does. Thank you! Cool!

**Comment by Andrei-Aksionov** [comment]
I think it's a bit early for me to mess around with Flash Attention kernel, but maybe a bit later 🤔.
Anyway, thanks @carmocca for the tip, and thanks @t-vi for the notebook 👍.

**Pull request merged**